Codigo/ModLobby.bas

```diff
@@ -0,0 +1,218 @@
+    instance.State = Initialized
+    instance.SummonCoordinates.map = -1
+End Sub
+
+Public Sub SetSummonCoordinates(ByRef instance As t_Lobby, ByVal map As Integer, ByVal posX As Integer, ByVal posY As Integer)
+    instance.SummonCoordinates.map = map
+    instance.SummonCoordinates.X = posX
+    instance.SummonCoordinates.y = posY
+End Sub
+
+Public Sub SetMaxPlayers(ByRef instance As t_Lobby, ByVal playerCount As Integer)
+    instance.MaxPlayers = playerCount
+    ReDim instance.Players(0 To playerCount - 1)
+End Sub
+
+Public Sub SetMinPlayers(ByRef instance As t_Lobby, ByVal playerCount As Integer)
+    instance.MinPlayers = playerCount
+End Sub
+
+Public Sub SetMinLevel(ByRef instance As t_Lobby, ByVal level As Byte)
+    instance.MinLevel = level
+End Sub
+
+Public Sub SetMaxLevel(ByRef instance As t_Lobby, ByVal level As Byte)
+    instance.MaxLevel = level
+End Sub
+
+Public Sub SetClassFilter(ByRef instance As t_Lobby, ByVal Class As Integer)
+    instance.ClassFilter = Class
+End Sub
+
+Public Function AddPlayer(ByRef instance As t_Lobby, ByVal UserIndex As Integer) As t_response
+On Error GoTo AddPlayer_Err
+    With UserList(UserIndex)
+        If .Stats.ELV < instance.MinLevel Or .Stats.ELV > instance.MaxLevel Then
+            AddPlayer.Success = False
+            AddPlayer.Message = 396
+            Exit Function
+        End If
+        If instance.RegisteredPlayers >= instance.MaxPlayers Then
+            AddPlayer.Success = False
+            AddPlayer.Message = 397
+            Exit Function
+        End If
+        If instance.ClassFilter > 0 And .clase <> instance.ClassFilter Then
+            AddPlayer.Success = False
+            AddPlayer.Message = 398
+            Exit Function
+        End If
+        Dim playerPos As Integer: playerPos = instance.RegisteredPlayers
+        instance.Players(playerPos).UserId = UserIndex
+        instance.Players(playerPos).IsSummoned = False
+        instance.RegisteredPlayers = instance.RegisteredPlayers + 1
+        AddPlayer.Message = 399
+        AddPlayer.Success = True
+        If instance.SummonAfterInscription Then
+            Call SummonPlayer(instance, playerPos)
+        End If
+    End With
+    Exit Function
+AddPlayer_Err:
+102     Call TraceError(Err.Number, Err.Description, "ModLobby.AddPlayer", Erl)
+
+End Function
+
+Public Sub SummonPlayer(ByRef instance As t_Lobby, ByVal user As Integer)
+   On Error GoTo SummonPlayer_Err
+        Dim UserIndex As Integer
+        With instance.Players(user)
+            UserIndex = .UserId
+            If Not .IsSummoned Then
+                .SummonedFrom = UserList(UserIndex).Pos
+            End If
+100         Call WarpToLegalPos(UserIndex, instance.SummonCoordinates.map, instance.SummonCoordinates.X, instance.SummonCoordinates.y, True, True)
+            .IsSummoned = True
+        End With
+    Exit Sub
+SummonPlayer_Err:
+102     Call TraceError(Err.Number, Err.Description, "ModLobby.SummonPlayer_Err", Erl)
```

Comment:
102 and 100 are repeated in the same file, this should give compiler errors. Please remove line numbers.

Reply:
these line numbers are on different functions, this builds fine, you can see more examples of this in protocol.bas